Bluetooth: ISO: Guard addition ISO debug with CONFIG_BT_DEBUG_ISO_DATA

Thalley opened the pull request:
Guard additional debug statements in ISO that deals with TX/RX with CONFIG_BT_DEBUG_ISO_DATA to avoid spamming the log.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: ISO: Guard addition ISO debug with CONFIG_BT_DEBUG_ISO_DATA

Guard additional debug statements in ISO that deals
with TX/RX with CONFIG_BT_DEBUG_ISO_DATA to avoid spamming
the log.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

alwa-nordic commented:
Have you looked at `LOG_INST_DBG`? I think it's exactly for this purpose. And it will allow setting up runtime filters. Runtime filters are nice when debugging because they can can be controlled programatically in e.g. a modified sample.

Thalley commented:
> Have you looked at `LOG_INST_DBG`? I think it can be used for this purpose, even though it it's meant for "instances". It will allow setting up runtime filters. Runtime filters are nice when debugging because they can can be controlled programatically in a modified sample to only capture logs in a small window.

I haven't :) So basically you can setup a logging instance for e.g. ISO data, and use that for this? 

I guess that could work, and is perhaps something we should to look into

cvinayak commented on subsys/bluetooth/host/iso.c:
```diff
@@ -93,7 +93,9 @@ void hci_iso(struct net_buf *buf)
 	uint8_t flags;
 
-	BT_DBG("buf %p", buf);
+	if (IS_ENABLED(CONFIG_BT_DEBUG_ISO_DATA)) {
+		BT_DBG("buf %p", buf);
+	}
```
To reduce noise, conditionally define a `BT_ISO_DATA_DBG`  in https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/common/log.h
```suggestion
	BT_ISO_DATA_DBG("buf %p", buf);
```

Thalley replied:
I'd argue that this is outside the scope of this PR. Several debug statements are already guarded by `CONFIG_BT_DEBUG_ISO_DATA`, and this PR simply fixes the remaining that wasn't added the first time. 

w.r.t your request, I think I'd need input from more, as I slightly disagree: IMO it wouldn't reduce noise, but rather readability.

cvinayak replied:
I could only find two early use of `CONFIG_BT_DEBUG_ISO_DATA` in upstream main branch. This PR adds 5 new additional conditional logging.

It is cleaner to use a macro instead of adding more lines of code for each conditional logging being done in this PR.

Thalley replied:
Alright. A similar request has been made for https://github.com/zephyrproject-rtos/zephyr/pull/50662, so I'll go ahead and add it :)

Thalley committed:
Bluetooth: ISO: Replace CONFIG_BT_DEBUG_ISO_DATA guard with BT_ISO_DATA_DBG

Instead of guarding all BT ISO data log statements with
CONFIG_BT_DEBUG_ISO_DATA, a new macro, BT_ISO_DATA_DBG, is
implemented to handle the guarding.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley commented:
Rebased to solve merge conflicts

alwa-nordic commented on subsys/bluetooth/common/log.h:
```diff
@@ -41,6 +41,12 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME, LOG_LEVEL);
 #define BT_WARN(fmt, ...) LOG_WRN(fmt, ##__VA_ARGS__)
 #define BT_INFO(fmt, ...) LOG_INF(fmt, ##__VA_ARGS__)
 
+#if defined(CONFIG_BT_DEBUG_ISO_DATA)
+#define BT_ISO_DATA_DBG(fmt, ...) BT_DBG(fmt, ##__VA_ARGS__)
+#else
+#define BT_ISO_DATA_DBG(fmt, ...)
+#endif /* CONFIG_BT_DEBUG_ISO_DATA */
+
```
This file `subsys/bluetooth/common/log.h` is going away. Could we put this directly in `iso.c`?

For context, PR's related to removal of `bluetooth/common/log.h`.
https://github.com/zephyrproject-rtos/zephyr/pull/51562
https://github.com/zephyrproject-rtos/zephyr/pull/51595

Thalley replied:
I am OK with that. @cvinayak Do you agree?

Thalley replied:
Updated with this suggestion

Thalley commented:
@cvinayak @kruithofa  Please re-review :)
